src/common/client-app.ts

```diff
@@ -20,44 +20,50 @@ export var defaultConfiguration: AppConfiguration = {
 };
 
 export function start(boardVM: IBoardVM, config?: AppConfiguration) {
-  var app = new App(boardVM, config);
-  app.start();
+  var app = new App(boardVM);
+  config = utils.extend({ }, defaultConfiguration, config);
+  boardVM.applyBindings(config.rootNode);
+  var socket = io();
+  socket.on(config.socketEventName, app.onMessage);
+
+  app.sendToServer = (changes) => {
+      socket.emit(config.socketEventName, { patch: changes });
+  };
+  setInterval(app.onInterval, config.throttlingInterval);
   return app;
 }
 
-class App {
-  shadowServer: model.Board = { };
+export class App {
   shadowClient: model.Board = { };
   boardVM: IBoardVM;
-  socket: SocketIOClient.Socket;
-  config: AppConfiguration;
+  sendToServer: (changes: any) => void;
 
   constructor(boardVM: IBoardVM, config?: AppConfiguration) {
     this.boardVM = boardVM;
-    this.config = utils.extend({ }, defaultConfiguration, config);
+    this.shadowClient = this.boardVM.toPlain();

```

Comment:
It's the same for me, considering that in a new refactoring we can continue using the a shadow board for server (or not, it's depends where we'll put it, server or client).